### PR TITLE
Tune GHA workflows

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

See commits

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ---
 name: CrateDB SQL
 on:
-  pull_request: ~
+  pull_request:
+    paths:
+      - '**.java'
```
Comment (matriv) [inline .github/workflows/main.yml:6]: is this correct? what about changes in gradle files, dependencies, etc?

**Review by matriv — COMMENTED** [review]
Maybe I didn't understand correctly the change: blackbox tests are also running when java files are changed, but we don't run java tests when docs only have changed?

File: .github/workflows/main.yml (hunk 1 above)
Comment (mfussenegger) [reply .github/workflows/main.yml:6]: It wasn't. Added gradle as well

**Comment by mfussenegger** [comment]
> blackbox tests are also running when java files are changed, but we don't run java tests when docs only have changed?

- Everything we test on Jenkins runs always. (this includes itests and the blackbox stuff)
- Vale & linkcheck runs always
- GHA java tests only run if java or gradle code changed

File: .github/workflows/main.yml (hunk 1 above)
Comment (matriv) [reply .github/workflows/main.yml:6]: I'd say that it's not enough, what about `version.properties`, or changes in files under `resources`?

Comment (mfussenegger) [reply .github/workflows/main.yml:6]: Okay, added `version.properties` too. I don't think it's worth making this more complicated than that. The intention is to make this work for the common cases, we still have Jenkins which always runs the full test suite anyways.

**Review by matriv — APPROVED** [review]
Thx for the improvement and explanation @mfussenegger!

**Pull request merged**